### PR TITLE
Enhance error handling for IPv4 and IPv6 conversion functions

### PR DESCRIPTION
**Pull request by vidhvath28**
I've implemented significant enhancements to the error handling in the IPv4 and IPv6 conversion functions. Here are the key changes:

Custom Error Classes: Created specific error classes for invalid IP addresses and type mismatches to improve clarity in error reporting.
Detailed Validation: Added comprehensive input validation to ensure that only valid IPv4 and IPv6 addresses are processed.
Consistent Type Checking: Improved the consistency of type checking for function inputs to avoid unexpected behavior.

### CODE REVIEW

File: index.spec.ts [hunk 1]
```diff
@@ -76,5 +115,51 @@ test("should be 21da:d4::2f4c:2bc:ff:fe18:4c5a", () => {
...
+
+test("should throw error for incomplete IPv6 strings", () => {
+  expect(() => fromIPv6("2001:4860")).toThrow();
+  expect(() => fromIPv6("::1::")).toThrow();
+});
+
   expect(toIPv6(42541956123769884636017138956568135816n)).toBe("2001:4860:4860::8888");
-});
\ No newline at end of file
+});
```
Comment (0xflotus) [inline index.spec.ts:164]: This should be deleted. Maybe you had some problems with a merge conflict?

**Comment by 0xflotus** [comment]
This is a duplicate of #680, right?

**Comment by vidhvath28** [comment]
> This is a duplicate of #680, right?

I wanted the keep the PR specifically for two different task one for Test cases and one for error handling so, I did submit 2 different PR , it would be great if you merge this PR also .

**Comment by 0xflotus** [comment]
The changes of this PR are committed. So this PR won't affect the code. 

Your PR #680 closed the two issues #659 and #661

**Pull request closed without merging**